src/main/java/ui/components/FilterTextField.java

```diff
@@ -94,6 +100,16 @@ private void setup() {
     }
 
     /**
+     * Sets the background of FilterTextField to the appropriate color. 
+     * The color set depends on isValidFilter.
```

Comment:
Alternatively (preferred), you can have two methods. `setStyleForValidFilter()` and `setStyleForInvalidFilter()`.
The call site will be more readable that way.